go/private/BUILD.sdk.bazel

```diff
@@ -80,6 +80,7 @@ filegroup(
         "bin/go*",
         "src/**",
         "pkg/**",
+        {goenv}
```

Comment:
`go.env*` should match `go.env` if it exists and nothing otherwise. Just slightly hacky, but in my opinion better than version detection logic. I could see a point for `*.env` instead.

Reply:
Nice call. I switched to using `go.env*` instead, which is much cleaner.

Let's start with the more conservative solution and change it later when needed.

Reply:
```
(09:55:02) ERROR: Traceback (most recent call last):
--
  | File "/var/lib/buildkite-agent/.cache/bazel/_bazel_buildkite-agent/ec321eb2cc2d0f8f91b676b6d4c66c29/external/go_sdk/BUILD.bazel", line 79, column 16, in <toplevel>
  | srcs = glob([
  | Error in glob: glob pattern 'go.env*' didn't match anything, but allow_empty is set to False (the default value of allow_empty can be set with --incompatible_disallow_empty_glob).
```

I guess it does not work if empty glob is not allowed.

Reply:
You can set `allow_empty = True`

Reply:
You can also extract out a separate `glob` call so that we don't lose the error in case the SDK directory is empty.